### PR TITLE
(maint) Changes to support TLS 1.3

### PR DESCRIPTION
**Pull request by Magisus**

### CODE REVIEW

File: .travis.yml
Comment (mwaggett) [inline .travis.yml:15]: wondering if all these should go into a script that can take the Java version as a param, so we don't have to repeat it four times.

Comment (Magisus) [reply .travis.yml:15]: Yeah probably. It does look like they're identical except for the number.

Comment (justinstoller) [reply .travis.yml:15]: Is it possible/desirable to just run `sudo apt-get install -y openjdk-8-jdk` without first deleting the old java?

Comment (Magisus) [reply .travis.yml:15]: I can try it. I copied this from what Nick did in [clj-http-client](https://github.com/puppetlabs/clj-http-client/commit/6a6f27f7be96fbcc8479fd352952476b496925c8). If it works we can do that there too I suppose.

Comment (Magisus) [reply .travis.yml:15]: Apparently not 🤷‍♀️

**Comment by Magisus** [comment]
So I don't know why, but moving into a script caused it to try to download a different version, that doesn't exist:
```
Err:1 http://security.ubuntu.com/ubuntu bionic-security/universe amd64 openjdk-8-jre-headless amd64 8u292-b10-0ubuntu1~18.04
  404  Not Found [IP: 91.189.91.38 80]
Err:2 http://security.ubuntu.com/ubuntu bionic-security/universe amd64 openjdk-8-jdk-headless amd64 8u292-b10-0ubuntu1~18.04
  404  Not Found [IP: 91.189.91.38 80]
E: Failed to fetch http://security.ubuntu.com/ubuntu/pool/universe/o/openjdk-8/openjdk-8-jre-headless_8u292-b10-0ubuntu1~18.04_amd64.deb  404  Not Found [IP: 91.189.91.38 80]
E: Failed to fetch http://security.ubuntu.com/ubuntu/pool/universe/o/openjdk-8/openjdk-8-jdk-headless_8u292-b10-0ubuntu1~18.04_amd64.deb  404  Not Found [IP: 91.189.91.38 80]
```
Compared to just listing it in `before_install`:
```
Get:1 http://archive.ubuntu.com/ubuntu bionic-updates/universe amd64 openjdk-8-jre-headless amd64 8u312-b07-0ubuntu1~18.04 [28.2 MB]
Get:2 http://archive.ubuntu.com/ubuntu bionic-updates/universe amd64 openjdk-8-jdk-headless amd64 8u312-b07-0ubuntu1~18.04 [8,298 kB]
Fetched 36.5 MB in 3s (14.5 MB/s)
```

**Comment by Magisus** [comment]
@nmburgan when you were looking into updating java for TLS 1.3 in the http-client library, did you find anything about this, or did you not try anything besides using `before_install`?

**Comment by nmburgan** [comment]
I want to say `before_install` happens before in does `lein` stuff.  Don't know why it'd be looking in a different repo, though.  

Only other thing I looked at was using https://github.com/sormuras/bach/blob/master/install-jdk.sh, but you have to pass it your own URL since the ones it has baked in are old, and it didn't seem like much of an improvement.

**Comment by nmburgan** [comment]
Oh, also, originally I had it in the `script` section, but that required running `lein deps` again.  So `before_install` was better.

**Comment by Magisus** [comment]
This PR should be rebased once the new clj-parent is out and promoted.

**Review by mwaggett — APPROVED** [review]

**Comment by mwaggett** [comment]
This was partially superseded by https://github.com/puppetlabs/puppetserver/pull/2616 and the rest is blocked on https://github.com/puppetlabs/clj-parent/pull/314, so gonna close this and we can figure out what travis changes are needed once the clj-parent PR lands.

**Comment by mwaggett** [comment]
The rest of this is now included in https://github.com/puppetlabs/puppetserver/pull/2639.

**Pull request closed without merging**
